app/access/service_access.rb

```diff
@@ -27,8 +27,7 @@ def update?(_object, _params=nil)
     end
 
     def index?(_object_class, _params=nil)
-      # This can return true because the index endpoints filter objects based on user visibilities
-      true
+      admin_user? || admin_read_only_user? || has_read_scope? || global_auditor? || !VCAP::CloudController::FeatureFlag.enabled?(:hide_marketplace_from_unauthenticated_users)
```

Comment:
I'm not super excited about including business logic here, but there is some prior art for checking feature flags in the v2 access classes: https://github.com/search?q=repo%3Acloudfoundry%2Fcloud_controller_ng+path%3A%2F%5Eapp%5C%2Faccess%5C%2F%2F+FeatureFlag&type=code

If this logic needs to go here, can/should we remove the corresponding logic in the controller?https://github.com/cloudfoundry/cloud_controller_ng/blob/7b1848e6b3d33ecc486adb55fbbe35fe44fb2cb3/app/controllers/services/services_controller.rb#L28-L39

Reply:
> If this logic needs to go here, can/should we remove the corresponding logic in the controller?

possibly? Do you want me to?

I'd rather there not be that code in the access controller at all tbh, but i would have done that by just, never anymore allowing unauthorized users to access the marketplace endpoint. Given we have what I'd, honestly consider a "compatibility flag" more then a "feature flag"(as I mentioned in my comment), I figured I should try to respect that flag unless we're thinking we're ok with dropping it.

Reply:
Is it ever possible for the services controller code to execute? If the `InvalidAuthToken` and `InvalidAuthToken` cases are already handled elsewhere, then I think we should delete the vestigial code.

Reply:
I'll try and look into this and remove the redundancy.

Reply:
hmmm. I did "quick" checks. I'll do more tomorrow. It appears to be not possible to quickly remove.

Reply:
I'm doing research, It seems like service and service_plan access effect /v2/service calls, but weirdly enough not /v2/service_plan calls?